src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -2,6 +2,472 @@
+
+<Alert type='info'>
+
+If you have installed the toolchain and you are still getting errors like the
+one mentioned above, try running `sudo ldconfig` on the robot.
+
+</Alert>
+
+##### Module
+
+The `module` command allows you to generate a new module. To generate a new line detection module one would run
+
+```bash
+./b module generate vision/LineDetector
+```
+
+This will generate all necessary module files in the `module/vision/LineDetector` folder. This includes an empty configuration file, a basic C++ source and header file implementing the `LineDetector` reactor, and an empty unit test file.
+
+##### NBS
+
+NBS (NUClear Binary Stream) files are, effectively, concatenations of protobuf messages with a small header packet per message. They provide a simple and effective means for recording real data from the robots in real time.
+
+The `nbs` command provides a couple of subcommands to help you work with NBS data.
+
+| Subcommand     | Description                                                                              |
+| -------------- | ---------------------------------------------------------------------------------------- |
+| extract_images | Extracts images from the NBS file and save them to disk                                  |
+| json           | Extracts selected messages from the NBS file and dumps them to `stdout` in a json format |
+| stats          | Calculates message statistics on the provided NBS file                                   |
+| video          | Extract images from the NBS file and save them in a video file using FFmpeg              |
+
+##### Run
+
+To run roles locally on your system use the `run` command of `b` as follows
+
+```bash
+./b run <name of role>
+```
+
+<Alert type="warning">
+
+Be sure that you are using the `generic` target for this (`./b target generic`), otherwise your role will likely crash (unless your CPU happens to be the same or newer than the 7th generation Core i7 that the robots are currently using).
+
+</Alert>
+
+##### Shell
+
+This is an undocumented and unsupported option.
+
+<details>
+  <summary>I know what I am doing</summary>
+
+You probably don't know what you are doing. But why would you listen to me?
+
+For some reason, if you really, really, _REALLY_ need to access a terminal inside of Docker you can run
+
+```bash
+./b shell
+```
+
+<Alert type="warning">
+
+Any changes you make outside of the `/home/nubots/NUbots` or `/home/nubots/build` directories will not persist after exiting the Docker shell.
+
+</Alert>
+
+</details>
+
+##### Target
+
+The `target` command of `b` allows you to build code for a different target. Currently, only two targets are supported: `generic` and `nuc7i7bnh`.
```

Comment:
Didn't we add the nuc8 as well? or was that in a PR still

Reply:
nuc8 is still a PR, we can add that in a later PR as well